File: evidence/generated-systems/golang/crm-on-golang/crm-on-golang/internal/model/Account.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// Account Declaration
//==============================================================
type Account struct {
    gorm.Model
     Name                                    string
    AccountNumber                                    string
    Industry                                    string
    BillingAddress                                                            string
    ShippingAddress                                                            string
    Website                                                            string
    Phone                                                            string
    AsActive                                    bool
    OrganizationId         *uint
    Organization           *Organization `gorm:"foreignKey:OrganizationId"`
    ParentAccountId         *uint
    ParentAccount           *Account `gorm:"foreignKey:ParentAccountId"`
     ChildAccounts           []Account `gorm:"foreignKey:ChildAccountsFromAccountId"`
     Contacts           []Contact `gorm:"foreignKey:ContactsFromAccountId"`
     Opportunities           []Opportunity `gorm:"foreignKey:OpportunitiesFromAccountId"`
     Cases           []Case_ `gorm:"foreignKey:CasesFromAccountId"`
    OwnerId         *uint
    Owner           *User `gorm:"foreignKey:OwnerId"`
    TerritoryId         *uint
    Territory           *Territory `gorm:"foreignKey:TerritoryId"`
     Activities           []Activity `gorm:"foreignKey:ActivitiesFromAccountId"`
     Campaigns           []Campaign `gorm:"foreignKey:CampaignsFromAccountId"`
     Quotes           []Quote `gorm:"foreignKey:QuotesFromAccountId"`
     Orders           []Order `gorm:"foreignKey:OrdersFromAccountId"`
     Contracts           []Contract `gorm:"foreignKey:ContractsFromAccountId"`
     Notes           []Note `gorm:"foreignKey:NotesFromAccountId"`
     EmailMessages           []EmailMessage `gorm:"foreignKey:EmailMessagesFromAccountId"`
    AccountType                      AccountType
    LifecycleStage                      AccountLifecycleStage

// parent associations as their child

}

File: evidence/generated-systems/rubyonrails/crm-on-rails/crmonrails/app/models/account.rb
class Account < ApplicationRecord
  enum AccountType: [:Prospect, :Customer, :Partner, :Vendor, :Competitor]
  enum LifecycleStage: [:Subscriber, :Lead, :MarketingQualified, :SalesQualified, :Customer, :Evangelist, :Churned]


  composed_of :address,
    class_name: "Address",
    mapping: [
      ${$mapping}, 
      ${$mapping}, 
      ${$mapping}, 
      ${$mapping}, 
      %w[address_country country]
    ]

  composed_of :address,
    class_name: "Address",
    mapping: [
      ${$mapping}, 
      ${$mapping}, 
      ${$mapping}, 
      ${$mapping}, 
      %w[address_country country]
    ]

  composed_of :uRL,
    class_name: "URL",
    mapping: [
      %w[uRL_value value]
    ]

  composed_of :phoneNumber,
    class_name: "PhoneNumber",
    mapping: [
      ${$mapping}, 
      ${$mapping}, 
      %w[phoneNumber_extension extension]
    ]

  has_many :Organization, class_name: 'Organization'
  has_many :ParentAccount, class_name: 'Account'
  has_many :ChildAccounts, class_name: 'Account'
  has_many :Contacts, class_name: 'Contact'
  has_many :Opportunities, class_name: 'Opportunity'
  has_many :Cases, class_name: 'Case_'
  has_many :Owner, class_name: 'User'
  has_many :Territory, class_name: 'Territory'
  has_many :Activities, class_name: 'Activity'
  has_many :Campaigns, class_name: 'Campaign'
  has_many :Quotes, class_name: 'Quote'
  has_many :Orders, class_name: 'Order'
  has_many :Contracts, class_name: 'Contract'
  has_many :Notes, class_name: 'Note'
  has_many :EmailMessages, class_name: 'EmailMessage'

end
